atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/memory/InMemoryAtlasDbFactory.java

```diff
@@ -68,6 +69,8 @@ public String getType() {
 
     @Override
     public InMemoryKeyValueService createRawKeyValueService(KeyValueServiceConfig config) {
+        AtlasDbVersion.ensureVersionReported();
```

Comment:
Why did you only include the check in the InMemoryKVS factory and not the factories of the other KVSs?  My understanding is the goal of this is to report out the Atlas version in production environments?  In which case my hope would be that they are using the ServiceDiscoveringAtlasSupplier.